xla/service/gpu/gpu_fused_mha_runner.cc

```diff
@@ -256,6 +104,14 @@ Status RunGpuFMHAImpl(const GpufMHAParams &params, se::Stream *stream,
       params.activation_buffer.is_null()
           ? se::DeviceMemoryBase()
           : se::DeviceMemory<OutputType>(params.activation_buffer);
+  auto mask_buffer =
```

Comment:
Comment by @bchetioui 

The removal of std::optional for the mask and bias buffers seems like a footgun.

Reply:
I agreed we should add back optional to make the semantics more clear. Should also add back for activation and other optional buffers as well.

Reply:
yea, let's keep them consistent with lhlo call interface(which we didnt do a good job the first time :) )

Reply:
Can you please push another version of this PR where the std::optional is used again?

Reply:
I pushed another version to use std::optional